tests/python_package_test/test_dask.py

```diff
@@ -1078,3 +1079,72 @@ def test_dask_methods_and_sklearn_equivalents_have_similar_signatures(methods):
     for param in dask_spec.args:
         error_msg = f"param '{param}' has different default values in the methods"
         assert dask_params[param].default == sklearn_params[param].default, error_msg
+
+
+@pytest.mark.parametrize('task', tasks)
+@pytest.mark.parametrize('output', data_output)
+@pytest.mark.parametrize('init_score', [-1, 1])
+def test_init_score(
+    task,
+    output,
+    init_score,
+    client,
+    listen_port
+):
+    if task == 'ranking' and output == 'scipy_csr_matrix':
+        pytest.skip('LGBMRanker is not currently tested on sparse matrices')
+
+    if task == 'ranking':
+        X, y, w, g, dX, dy, dw, dg = _create_ranking_data(
+            output=output,
+            group=None
+        )
+        dask_model_factory = lgb.DaskLGBMRanker
+        local_model_factory = lgb.LGBMRanker
+    else:
+        X, y, w, dX, dy, dw = _create_data(
+            objective=task,
+            output=output,
+        )
+        dg = None
+        if task == 'classification':
+            dask_model_factory = lgb.DaskLGBMClassifier
+            local_model_factory = lgb.LGBMClassifier
+        elif task == 'regression':
+            dask_model_factory = lgb.DaskLGBMRegressor
+            local_model_factory = lgb.LGBMRegressor
+    # training on multiple workers can lead to different results
+    # we make sure to train on only one worker to compare the results
+    dX = collection_to_single_partition(dX)
+    dy = collection_to_single_partition(dy)
+    dw = collection_to_single_partition(dw)
+    dg = collection_to_single_partition(dg)
```

Comment:
please use multi-worker training like the other tests, for the same reasons I gave in https://github.com/microsoft/LightGBM/pull/3943/files#r575609015 and https://github.com/microsoft/LightGBM/pull/3943/files#r575617861

If that means that `assert_eq()` doesn't quite work, it's enough to just check that the predictions are different when init_score is used vs. when it is not used.

Reply:
would this be something like fitting a model without init score and then fitting one with init score, computing the predictions for each one and doing `assert not np.allclose(regular_model_predictions, model_with_init_score_predictions)`?

Reply:
I mean that you can change the tolerances in `assert assert_eq(local_preds, dask_preds)` if you need to.

And then yeah, `assert not assert_eq(regular_model_predictions, model_with_init_score_predictions)` is probably good enough for now. Until we support `raw_score` and we can make a more direct comparison (#3793 )